### PR TITLE
Cmdlets for Cross server gt

### PR DESCRIPTION
**Pull request by adumitr**

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@adumitr__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (adumitr). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

**Comment by adumitr** [comment]
Is it at all possible to get this merged today? Also, after, please send me some details on how to get the changes published so they become publicly accessible (my alias is adumitr)

**Comment by markcowl** [comment]
@azuresdkci add to whitelist

**Comment by adumitr** [comment]
After another build fix and retrying a few builds to get flaky tests to pass, finally got a successful build. Can we complete this merge? Thanks

**Comment by adumitr** [comment]
what is the status with this pull request? It says checks have been pending for way too long, plus there had been a prior successful build that passed all checks. What does it take at this point to complete this?

**Comment by markcowl** [comment]
@azuresdkci retest this please

File: src/ResourceManager/Sql/Commands.Sql/ServerCommunicationLink/Cmdlet/GetAzureSqlServerCommunicationLink.cs [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+using System.Collections.Generic;
+using System.Management.Automation;
+using Microsoft.Azure.Commands.Sql.ServerCommunicationLink.Model;
+
+namespace Microsoft.Azure.Commands.Sql.ServerCommunicationLink.Cmdlet
+{
+    [Cmdlet(VerbsCommon.Get, "AzureRmSqlServerCommunicationLink", 
+        ConfirmImpact = ConfirmImpact.None)]
+    public class GetAzureSqlServerCommunicationLink : AzureSqlServerCommunicationLinkCmdletBase
```
Comment (markcowl) [inline src/ResourceManager/Sql/Commands.Sql/ServerCommunicationLink/Cmdlet/GetAzureSqlServerCommunicationLink.cs]: @adumitr What is the OutputType for this cmdlet?  Please add the attribute for OutputType with the cmdlet OutputType

File: src/ResourceManager/Sql/Commands.Sql/ServerCommunicationLink/Cmdlet/NewAzureSqlServerCommunicationLink.cs [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+using Microsoft.Azure.Commands.Sql.Server.Adapter;
+
+namespace Microsoft.Azure.Commands.Sql.ServerCommunicationLink.Cmdlet
+{
+    /// <summary>
+    /// Cmdlet to create a new Azure Sql server communication link
+    /// </summary>
+    [Cmdlet(VerbsCommon.New, "AzureRmSqlServerCommunicationLink",
+        ConfirmImpact = ConfirmImpact.Low)]
+    public class NewAzureSqlServerCommunicationLink : AzureSqlServerCommunicationLinkCmdletBase
```
Comment (markcowl) [inline src/ResourceManager/Sql/Commands.Sql/ServerCommunicationLink/Cmdlet/NewAzureSqlServerCommunicationLink.cs]: OutputType?

File: src/ResourceManager/Sql/Commands.Sql/ServerCommunicationLink/Cmdlet/RemoveAzureSqlServerCommunicationLink.cs [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+using System.Management.Automation;
+using Microsoft.Azure.Commands.Sql.ServerCommunicationLink.Model;
+using Microsoft.Azure.Commands.Sql.Properties;
+
+namespace Microsoft.Azure.Commands.Sql.ServerCommunicationLink.Cmdlet
+{
+    [Cmdlet(VerbsCommon.Remove, "AzureRmSqlServerCommunicationLink",
+        SupportsShouldProcess = true, 
+        ConfirmImpact = ConfirmImpact.High)]
+    public class RemoveAzureSqlServerCommunicationLink : AzureSqlServerCommunicationLinkCmdletBase
```
Comment (markcowl) [inline src/ResourceManager/Sql/Commands.Sql/ServerCommunicationLink/Cmdlet/RemoveAzureSqlServerCommunicationLink.cs]: OutputType?

File: src/ResourceManager/Sql/Commands.Sql/ServerCommunicationLink/Cmdlet/SetAzureSqlServerCommunicationLink.cs [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+using Microsoft.Azure.Commands.Sql.ServerCommunicationLink.Model;
+
+namespace Microsoft.Azure.Commands.Sql.ServerCommunicationLink.Cmdlet
+{
+    /// <summary>
+    /// Cmdlet to create a new Azure Sql Database server communication link
+    /// </summary>
+    [Cmdlet(VerbsCommon.Set, "AzureRmSqlServerCommunicationLink",
+        ConfirmImpact = ConfirmImpact.Medium)]
+    public class SetAzureSqlServerCommunicationLink : AzureSqlServerCommunicationLinkCmdletBase
```
Comment (markcowl) [inline src/ResourceManager/Sql/Commands.Sql/ServerCommunicationLink/Cmdlet/SetAzureSqlServerCommunicationLink.cs]: OutputType?

File: src/ResourceManager/Sql/Commands.Sql/ServerCommunicationLink/Services/AzureSqlServerCommunicationLinkCommunicator.cs [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+
+        /// <summary>
+        /// Gets or set the Azure subscription
+        /// </summary>
+        private static AzureSubscription Subscription { get; set; }
+
+        /// <summary>
+        /// Gets or sets the Azure profile
+        /// </summary>
+        public AzureContext Context { get; set; }
```
Comment (markcowl) [inline src/ResourceManager/Sql/Commands.Sql/ServerCommunicationLink/Services/AzureSqlServerCommunicationLinkCommunicator.cs]: Why do you have a separate Context property?  Context should always come from the Cmdlet context

File: src/ResourceManager/Sql/Commands.Sql/ServerCommunicationLink/Services/AzureSqlServerCommunicationLinkCommunicator.cs [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+    {
+        /// <summary>
+        /// The Sql client to be used by this end points communicator
+        /// </summary>
+        private static SqlManagementClient SqlClient { get; set; }
+
+        /// <summary>
+        /// Gets or set the Azure subscription
+        /// </summary>
+        private static AzureSubscription Subscription { get; set; }
```
Comment (markcowl) [inline src/ResourceManager/Sql/Commands.Sql/ServerCommunicationLink/Services/AzureSqlServerCommunicationLinkCommunicator.cs]: Why do you have a separate subscription property?  Subscription should come fromt eh cmdlet's context

File: src/ResourceManager/Sql/Commands.Sql/ServerCommunicationLink/Services/AzureSqlServerCommunicationLinkAdapter.cs [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+    {
+        /// <summary>
+        /// Gets or sets the AzureEndpointsCommunicator which has all the needed management clients
+        /// </summary>
+        private AzureSqlServerCommunicationLinkCommunicator Communicator { get; set; }
+
+        /// <summary>
+        /// Gets or sets the Azure profile
+        /// </summary>
+        public AzureContext Context { get; set; }
```
Comment (markcowl) [inline src/ResourceManager/Sql/Commands.Sql/ServerCommunicationLink/Services/AzureSqlServerCommunicationLinkAdapter.cs]: Same q, why the separate fields for context and subscription - this data should already be in the client, why keep it separately?

File: src/ResourceManager/Sql/Commands.Sql/ServerCommunicationLink/Services/AzureSqlServerCommunicationLinkCommunicator.cs (hunk 5 above)
Comment (adumitr) [inline src/ResourceManager/Sql/Commands.Sql/ServerCommunicationLink/Services/AzureSqlServerCommunicationLinkCommunicator.cs]: It does come from that, it gets passed from AzureSqlServerCommunicationLinkCmdletBase into the adapter and then into the communicator

File: src/ResourceManager/Sql/Commands.Sql/ServerCommunicationLink/Services/AzureSqlServerCommunicationLinkCommunicator.cs (hunk 6 above)
Comment (adumitr) [inline src/ResourceManager/Sql/Commands.Sql/ServerCommunicationLink/Services/AzureSqlServerCommunicationLinkCommunicator.cs]: It does come from that, it gets passed from AzureSqlServerCommunicationLinkCmdletBase into the adapter and then into the communicator

File: src/ResourceManager/Sql/Commands.Sql/ServerCommunicationLink/Services/AzureSqlServerCommunicationLinkAdapter.cs (hunk 7 above)
Comment (adumitr) [inline src/ResourceManager/Sql/Commands.Sql/ServerCommunicationLink/Services/AzureSqlServerCommunicationLinkAdapter.cs]: That just seems to be the pattern of the SQL cmdlets. I already passed code reviews in SQL, I'm not changing this

File: src/ResourceManager/Sql/Commands.Sql/ServerCommunicationLink/Cmdlet/SetAzureSqlServerCommunicationLink.cs (hunk 4 above)
Comment (adumitr) [inline src/ResourceManager/Sql/Commands.Sql/ServerCommunicationLink/Cmdlet/SetAzureSqlServerCommunicationLink.cs]: Added

File: src/ResourceManager/Sql/Commands.Sql/ServerCommunicationLink/Cmdlet/RemoveAzureSqlServerCommunicationLink.cs (hunk 3 above)
Comment (adumitr) [inline src/ResourceManager/Sql/Commands.Sql/ServerCommunicationLink/Cmdlet/RemoveAzureSqlServerCommunicationLink.cs]: Added

File: src/ResourceManager/Sql/Commands.Sql/ServerCommunicationLink/Cmdlet/NewAzureSqlServerCommunicationLink.cs (hunk 2 above)
Comment (adumitr) [inline src/ResourceManager/Sql/Commands.Sql/ServerCommunicationLink/Cmdlet/NewAzureSqlServerCommunicationLink.cs]: Added

File: src/ResourceManager/Sql/Commands.Sql/ServerCommunicationLink/Cmdlet/GetAzureSqlServerCommunicationLink.cs (hunk 1 above)
Comment (adumitr) [inline src/ResourceManager/Sql/Commands.Sql/ServerCommunicationLink/Cmdlet/GetAzureSqlServerCommunicationLink.cs]: Added

**Comment by adumitr** [comment]
Is anyone still around to merge this?

**Comment by stankovski** [comment]
@adumitr please fix merge conflicts

**Comment by adumitr** [comment]
Resolved conflicts. Does the merge work now?

**Pull request merged**